src/test/java/org/roaringbitmap/TestRoaringBitmap.java

```diff
@@ -4342,4 +4342,49 @@ public void testLazyOr() {
     Assert.assertEquals(rb2, rb1);
   }
 
+  @Test
+  public void testFirstLast_CreateSparseContainers() {
+    RoaringBitmap rb = new RoaringBitmap();
+    for(int i = 0; i < 20; ++i) {
+      int x = 1 << i;
+      rb.add(x);
+      Assert.assertEquals(1, rb.first());
+      Assert.assertEquals(x, rb.last());
+    }
+  }
+
+  @Test
+  public void testFirstLast_CreateSparseContainersAfterRun() {
+    RoaringBitmap rb = new RoaringBitmap();
+    rb.add(1L, 1 << 14);
+    for(int i = 18; i < 31; ++i) {
+      int x = 1 << i;
+      rb.add(x);
+      Assert.assertEquals(1, rb.first());
+      rb.last();
```

Comment:
Is this call mandatory ? You are not storing the last value and computing it again on the next line.